### PR TITLE
WIP/PEP8: pandas/core

### PR DESCRIPTION
**Pull request by rockg**
xref #11928 

Putting this out there as a start.  Will add to this over the next couple of days.

### CODE REVIEW

**Comment by jreback** [comment]
just merged: https://github.com/pydata/pandas/pull/11941

will show the linting effect in travis (though not fail because of it).

FYI, I have found ``autopep8`` does a pretty good job (though doesn't fix long lines)

**Comment by jreback** [comment]
I am also ok with doing this in sections (3 files sounds about right). So when one is ready to go. pls ping.

**Comment by rockg** [comment]
green...there are still a few E731 lambda should be def, but substantially better than before.

File: pandas/core/base.py [hunk 1]
```diff
@@ -345,7 +352,8 @@ def _gotitem(self, key, ndim, subset=None):
         """
         raise AbstractMethodError(self)
 
-    _agg_doc = """Aggregate using input function or dict of {column -> function}
+    _agg_doc = """Aggregate using input function or dict of {column ->
+    function}
```
Comment (jorisvandenbossche) [inline pandas/core/base.py]: There should not be a indentation here I think

File: pandas/core/base.py [hunk 2]
```diff
@@ -905,8 +923,10 @@ def drop_duplicates(self, keep='first', inplace=False):
         Parameters
         ----------
         keep : {'first', 'last', False}, default 'first'
-            - ``first`` : Mark duplicates as ``True`` except for the first occurrence.
-            - ``last`` : Mark duplicates as ``True`` except for the last occurrence.
+            - ``first`` : Mark duplicates as ``True`` except for the first
+            occurrence.
```
Comment (jorisvandenbossche) [inline pandas/core/base.py]: the 'occurence' should be aligned with '\`\`first\`\`'

File: pandas/core/base.py [hunk 3]
```diff
@@ -623,7 +634,8 @@ def __unicode__(self):
 
 
 class IndexOpsMixin(object):
-    """ common ops mixin to support a unified inteface / docs for Series / Index """
+    """ common ops mixin to support a unified inteface / docs for Series /
+    Index """
```
Comment (jorisvandenbossche) [inline pandas/core/base.py]: PEP8 says that if docstrings are not one-liners, the closing `"""` should be on its own line

**Comment by jorisvandenbossche** [comment]
@rockg can you specify what you exactly did? Is this the result of an automated tool (autopep8), or from manual clean-up?
Because I not fully 'happy' with all style changes, but if it is the result of a tool, then it will be probably easier to just accept the particular style of that tool (as style will always be something subjective).

Apart from that I put some comments of things that are not subjective.

**Comment by rockg** [comment]
Manual with the aid of editor defaults for line continuation, etc.  I'm happy to go through and change any persistent issues, but my main goal here was to result in no output from flake8.

**Comment by rockg** [comment]
As 90% of the issues are long-lines and autopep8 doesn't do that I felt it was easier to do manually.

**Comment by kawochen** [comment]
`--max-line-length 80` as an option to autopep8

**Comment by jreback** [comment]
that reminds me - we should recommend autopep8

any way to setup default options? like we do for flake8

**Comment by TomAugspurger** [comment]
Looks like it

```
  --ignore-local-config
                        don't look for and apply local config files; if not
                        passed, defaults are updated with any config files in
                        the project's root directory
```

it seem to look for a file called `.pep8` https://github.com/hhatto/autopep8/blob/9eb1121f357077c7d71fc770e25d3678f906a401/autopep8.py#L122

**Comment by rockg** [comment]
Yes, seems that it does.  I was only going on the comment above.  Is it in such a state to use autopep8 for this set or only do that for future ones?

**Comment by rockg** [comment]
For eventual continuity with the rest of the source, I'll use autopep8 and repush.

**Comment by jreback** [comment]
can u add in the config file for autopep8 as well

**Comment by jreback** [comment]
thanks @rockg

**Pull request merged**